src/main/groovy/com/morpheusdata/xen/XenserverProvisionProvider.groovy

```diff
@@ -1742,6 +1742,7 @@ class XenserverProvisionProvider extends AbstractProvisionProvider implements Wo
 					if (networkResults.success == true) {
 						def newInterface = buildNetworkInterface(computeServer, networkResults, newNetwork, newIndex, index)
 						newInterface.uniqueId = isWorkload ? "morpheus-nic-${instanceId}-${workload.id}-${newIndex}" : "morpheus-nic-${server.id}-${newIndex}"
+						newInterface.primaryInterface = networkAdd?.network?.isPrimary ? true : false
```

Comment:
I don't think `isPrimary` exists and `networkAdd.network` is the network associated to the interface, so it wouldn't have information to determine if the interface should be primary. 

A primary interface can't be removed or a new primary interface added, so `newInterface.primaryInterface` will always be false.

Reply:
review comment changes added